Mathlib/Algebra/Module/LocalizedModule.lean

```diff
@@ -577,6 +577,66 @@ lemma IsLocalizedModule.eq_iff_exists [IsLocalizedModule S f] {x₁ x₂} :
     apply_fun f at h
     simp_rw [f.map_smul_of_tower, Submonoid.smul_def, ← Module.algebraMap_end_apply R R] at h
     exact ((Module.End_isUnit_iff _).mp <| map_units f c).1 h
+
+theorem IsLocalizedModule.of_linearEquiv (e : M' ≃ₗ[R] M'') [hf : IsLocalizedModule S f] :
+    IsLocalizedModule S (e ∘ₗ f : M →ₗ[R] M'') where
+  map_units s := by
+    rw [show algebraMap R (Module.End R M'') s = e ∘ₗ (algebraMap R (Module.End R M') s) ∘ₗ e.symm
+      by ext; simp, Module.End_isUnit_iff, LinearMap.coe_comp, LinearMap.coe_comp,
+      LinearEquiv.coe_coe, LinearEquiv.coe_coe, EquivLike.comp_bijective, EquivLike.bijective_comp]
+    exact (Module.End_isUnit_iff _).mp <| hf.map_units s
+  surj' x := by
+    obtain ⟨p, h⟩ := hf.surj' (e.symm x)
+    exact ⟨p, by rw [LinearMap.coe_comp, LinearEquiv.coe_coe, Function.comp_apply, ← e.congr_arg h,
+      Submonoid.smul_def, Submonoid.smul_def, LinearEquiv.map_smul, LinearEquiv.apply_symm_apply]⟩
+  exists_of_eq h := by
+    simp_rw [LinearMap.coe_comp, LinearEquiv.coe_coe, Function.comp_apply,
+      EmbeddingLike.apply_eq_iff_eq] at h
+    exact hf.exists_of_eq h
+
+variable (M) in
+lemma isLocalizedModule_id (R') [CommRing R'] [Algebra R R'] [IsLocalization S R'] [Module R' M]
+    [IsScalarTower R R' M] : IsLocalizedModule S (.id : M →ₗ[R] M) where
+  map_units s := (Module.End_isUnit_iff _).mpr <| by
+    convert_to Function.Bijective (algebraMap R' (Module.End R' M) (algebraMap R R' s))
+    · ext; simp
+    exact (Module.End_isUnit_iff _).mp ((IsLocalization.map_units R' s).map _)
+  surj' m := ⟨(m, 1), one_smul _ _⟩
+  exists_of_eq h := ⟨1, congr_arg _ h⟩
+
+theorem isLocalizedModule_iff_isLocalization {A Aₛ} [CommRing A] [Algebra R A][CommRing Aₛ]
```

Comment:
I wonder if we should make the inverse direction an instance.

Reply:
I can try and see what happens but somebody with a better understanding of instances should advise about the pros and cons of this change.

Reply:
You can just try and bench it.

Reply:
So it looks fine: 
>  There were [no significant changes](http://speed.lean-fro.org/mathlib4/compare/32a979f1-1135-41e1-b764-f51014328bed/to/0cb17a0c-759e-4a34-8749-86bd7a0f459d) against commit https://github.com/leanprover-community/mathlib4/commit/95f91f8e66f14c0eecde8da6dbfeff39d44cbd81.